### PR TITLE
Update Error Reporting, Monitoring, and Trace docs

### PR DESCRIPTION
**Pull request by quartzmo**
This PR:

* Includes GAPIC types in JSON docs
* Updates TOCs with the design for GAPIC types in GoogleCloudPlatform/gcloud-common/issues/207
* Adds hand-authored tables of GAPIC "Data Types" per the design for GAPIC types in GoogleCloudPlatform/gcloud-common/issues/207

/cc @omaray

### CODE REVIEW

**Comment by quartzmo** [comment]
Bump! @dazuma / @hxiong388, look OK to you?

**Comment by hxiong388** [comment]
Looks so much better than current design!

File: google-cloud-error_reporting/lib/google/cloud/error_reporting/v1beta1/doc/google/devtools/clouderrorreporting/v1beta1/common.rb [hunk 1]
```diff
@@ -15,6 +15,34 @@
 module Google
   module Devtools
     module Clouderrorreporting
+      ##
+      # The `Google::Devtools::Clouderrorreporting::V1beta1` module provides the following types:
+      #
+      # Class | Description
+      # ----- | -----------
+      # {Google::Devtools::Clouderrorreporting::V1beta1::DeleteEventsRequest} | Deletes all events in the project.
```
Comment (dazuma) [inline google-cloud-error_reporting/lib/google/cloud/error_reporting/v1beta1/doc/google/devtools/clouderrorreporting/v1beta1/common.rb]: Most of the request object descriptions are of the form "specifies abc". This one is an odd one out, and a bit confusing: the object itself doesn't delete events; it's a request object for an API call that deletes events. Can we standardize the language a bit?

Comment (quartzmo) [inline google-cloud-error_reporting/lib/google/cloud/error_reporting/v1beta1/doc/google/devtools/clouderrorreporting/v1beta1/common.rb]: @dazuma. These descriptions are copy/paste from [the GAPIC source](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-error_reporting/lib/google/cloud/error_reporting/v1beta1/doc/google/devtools/clouderrorreporting/v1beta1/error_stats_service.rb#L238). (It's manual right now, but Song opened #1180 for generating the table.)

Comment (quartzmo) [inline google-cloud-error_reporting/lib/google/cloud/error_reporting/v1beta1/doc/google/devtools/clouderrorreporting/v1beta1/common.rb]: @swcloud See @dazuma's comment above on GAPIC docs.

Comment (swcloud) [inline google-cloud-error_reporting/lib/google/cloud/error_reporting/v1beta1/doc/google/devtools/clouderrorreporting/v1beta1/common.rb]: This should be a comment fix from the API provider.

**Pull request merged**